src/hotspot/share/gc/g1/g1AllocRegion.cpp

```diff
@@ -43,16 +43,19 @@ void G1AllocRegion::setup(G1CollectedHeap* g1h, HeapRegion* dummy_region) {
 
   // Make sure that any allocation attempt on this region will fail
   // and will not trigger any asserts.
-  assert(dummy_region->allocate_no_bot_updates(1) == NULL, "should fail");
-  assert(dummy_region->allocate(1) == NULL, "should fail");
   DEBUG_ONLY(size_t assert_tmp);
   assert(dummy_region->par_allocate_no_bot_updates(1, 1, &assert_tmp) == NULL, "should fail");
-  assert(dummy_region->par_allocate(1, 1, &assert_tmp) == NULL, "should fail");
 
   _g1h = g1h;
   _dummy_region = dummy_region;
 }
 
+void G1AllocRegion::update_bot_for_region_waste(HeapWord* addr, size_t size) {
+  assert(_alloc_region != NULL, "invariant");
+  assert(_bot_updates, "must only be called for regions doing BOT updates");
+  _alloc_region->update_bot_at(addr, size);
+}
+
```

Comment:
May I ask why do we need to update for the filler objects? Because we are not scanning them.

Reply:
Good question. You are correct in that we never need to scan these parts because of dirty cards, but during remembered set rebuilding (_see_: `G1RebuildRemSetHeapRegionClosure::rebuild_rem_set_in_region(...)`) we include the whole region when looking for references to other heap regions. 

There might be some good way to avoid scanning those parts during rebuild, but such investigation is out of scope for this PR.

Thanks for reviewing 😄